app/js/controllers/foyer/suggestion.js

```diff
@@ -79,31 +67,71 @@ angular.module('ddsApp').controller('SuggestionCtrl', function($scope, $http,dro
         delete expectedResult.shouldCompute;
     };
 
+    $scope.determineExtensionAndRepository = function(aids) {
```

Comment:
I don't see no particular reason for binding this to `$scope`, it looks like it is like this just to make it exposed to tests. If this is the reason, then I think this should be extracted to a `SuggestionService`